openslides/assignments/static/js/assignments/site.js

```diff
@@ -471,27 +499,21 @@ angular.module('OpenSlidesApp.assignments.site', ['OpenSlidesApp.assignments'])
     'Assignment',
     'AssignmentForm',
     'Agenda',
-    function($scope, Assignment, AssignmentForm, Agenda) {
+    'AgendaUpdate',
+    'AgendaTree',
+    function($scope, Assignment, AssignmentForm, Agenda, AgendaUpdate, AgendaTree) {
         $scope.model = {};
         // set default value for open posts form field
         $scope.model.open_posts = 1;
         // get all form fields
         $scope.formFields = AssignmentForm.getFormFields();
-
         // save assignment
         $scope.save = function(assignment) {
             Assignment.create(assignment).then(
                 function(success) {
-                    // find related agenda item
-                    Agenda.find(success.agenda_item_id).then(function(item) {
-                        // check form element and set item type (AGENDA_ITEM = 1, HIDDEN_ITEM = 2)
-                        var type = assignment.showAsAgendaItem ? 1 : 2;
-                        // save only if agenda item type is modified
-                        if (item.type != type) {
-                            item.type = type;
-                            Agenda.save(item);
-                        }
-                    });
+                    var changes = [{key: 'type', value: (assignment.showAsAgendaItem ? 1 : 2)},
```

Comment:
Please add a short comment like this:
`// Value 1 means a non hidden agenda item, value 2 means a hidden agenda item, see openslides.agenda.models.Item.ITEM_TYPE.`

You can choose a similar text if you like.

Please add the note to all other places where these changes arrays are build.